src/tools.cc

```diff
@@ -972,11 +972,11 @@ parseEtcHosts(void)
             debugs(1, 5, "etc_hosts: got hostname '" << lt << "'");
 
             /* For IPV6 addresses also check for a colon */
-            if (Config.appendDomain && !strchr(lt, '.') && !strchr(lt, ':')) {
+            if (Config.appendDomain && strlen(Config.appendDomain)<sizeof(buf2)-2 && !strchr(lt, '.') && !strchr(lt, ':')) {
```

Comment:
In latest Squid we have a helper function to handle this append directive more safely.

```suggestion
            (void)urlAppendDomain(lt);
```